### DISCUSSION TITLE
Question: Want to use my zshrc from source control

### DISCUSSION THREAD

**Discussion by wonbyte** [zimfw/zimfw#120] [Q&A] [answered]
Please check the existing issues to make sure you're not duplicating a report.

For bug reports, please provide the following information:

In a terminal, run `zmanage info` and paste the output below:

```
#Paste the output here
```

- gist of .zimrc:
- gist of .zshrc:

Description
-----------
Really love zim (switched from prezto). I want to use my .zshrc from git so is it ok for me to symlink the file to ~/.zshrc? I don't see anything special in the zshrc after doing the install so I assume this is ok. Should I then run zcompile on the .zshrc?

Also, is there Eriner theme using a custom font? 

Thanks for the help!

Steps to reproduce
------------------

Images or other information
---------------------------

**Comment by Plloi** [comment]
Personally my zimrc is in git but my dotfile git repo is based directly in my home dir so no symlinking necessary.   
That said, symlinking your .zshrc from a separate dotfiles dir shouldn't cause any issues.

**Comment by wonbyte** [comment]
I put the zlogin, zshrc, zimrc file under my home files source control. The only question I have is does the update feature of zim modify these?

Also is the 'Eriner' theme using a custom font? It does not seem to be displaying correctly in my prompt.

**Comment by Plloi** [comment]
presently zmanage update just pulls latest from git into ${ZDOTDIR} and updates submodules

**Comment by Eriner** [comment]
@wonbyte read the comments at the top of the theme source file, it has instructions for installing a font patch easily.

**Comment by Eriner** [comment, marked as answer]
> I don't see anything special in the zshrc after doing the install so I assume this is ok. Should I then run zcompile on the .zshrc?

@wonbyte running `source ~/.zlogin` is probably your best bet. This will zcompile your .zshrc and all other 'heavy' scripts that benefit from bytecode compilation.

**Comment by wonbyte** [comment]
@Eriner should I do it in the 

```
if [[ -s ${ZDOTDIR:-${HOME}}/.zim/init.zsh ]]; then
  source ${ZDOTDIR:-${HOME}}/.zim/init.zsh
fi
```

or after?

**Comment by Eriner** [comment]
@wonbyte you should only have to do it every once in a while, no reason to source it upon every shell invocation. Currently, the `.zlogin` file is sourced when you enter a TTY (which if you're starting X from a TTY, should mean it's sourced once on every login/boot).

**Comment by wonbyte** [comment]
thanks @Eriner . Love zim!

**Comment by Eriner** [comment]
No problem, glad you like it :)
